tokio/src/runtime/metrics/runtime.rs

```diff
@@ -127,6 +128,49 @@ impl RuntimeMetrics {
             self.handle.inner.num_idle_blocking_threads()
         }
 
+        /// Returns the thread id of the given worker thread.
+        ///
+        /// The returned value is `None` if the worker thread has not yet finished
+        /// starting up.
+        ///
+        /// If additional information about the thread, such as its native id, are
+        /// required, those can be collected in [`on_thread_start`] and correlated
+        /// using the thread id.
```

Comment:
This is enough for your purposes?

Reply:
Yes, for now we just want to send SIGABRT to the hanging thread and for that we collect the `pthread_t` id of each worker thread in `on_thread_start`.